nnvm/python/nnvm/frontend/tensorflow.py

```diff
@@ -469,6 +469,224 @@ def _impl(inputs, attr, params):
+                    next_index = min(input_dim - (stride_dim - i) + \
+                                     1 + num_add_axis_after_ellipsis, input_dim)
+                    for full_index in range(full_index, next_index):
+                        begin[full_index] = 0
+                        end[full_index] = inputs_0_shape[0][full_index]
+                        stride[full_index] = 1
+                        final_shape_gather_indices.append(full_index)
+
+                elif (1 << i) & new_axis_mask_inp:
+                    final_shape_gather_indices.append(kNewAxis)
+                else:
+                    if full_index == len(begin):
+                        return
+
+                    if begin_mask_inp & (1 << i):
+                        begin[full_index] = inputs_0_shape[0][full_index] \
+                            if stride_input[i] < 0 else 0
+                    elif begin_input:
+                        begin[full_index] = begin_input[i]
+
+                    if end_mask_inp & (1 << i):
+                        end[full_index] = 0 \
+                            if stride_input[i] < 0 else inputs_0_shape[0][full_index]
+                    elif end_input:
+                        end[full_index] = end_input[i]
+
+                    stride[full_index] = stride_input[i]
+
+                    if shrink_axis_mask_inp & (1 << i):
+                        final_shape_gather_indices.append(kShrinkAxis)
+                        begin[full_index] = \
+                            inputs_0_shape[0][full_index] + begin[full_index] \
+                                if begin[full_index] < 0 else begin[full_index]
+                        end[full_index] = begin[full_index] + 1
+                        stride[full_index] = 1
+                    else:
+                        final_shape_gather_indices.append(full_index)
+
+                    full_index = full_index + 1
+
+        transform_ellipsis()
+        out = _sym.strided_slice(inputs[0], begin=begin, end=end, stride=stride)
+        out_0_shape = _infer_out_shapes(out, params)
+        final_shape = []
+        for index in final_shape_gather_indices:
+            if index >= 0:
+                final_shape.append(out_0_shape[0][index])
+            elif index == kNewAxis:
+                final_shape.append(1)
+        outshape = tuple(final_shape)
+        return _sym.reshape(out, shape=outshape)
+    return _impl
+
+
+def _LSTMBlockCell():
+    def _impl(inputs, in_state_c, in_state_h, attr, params):
+        """LSTM Block cell.
+
+        ixh = [x, h_prev]
+        [i, ci, f, o] = xh * w + b
+        f = f + forget_bias
+
+        if not use_peephole:
+        wci = wcf = wco = 0
+
+        i = sigmoid(cs_prev * wci + i)
+        f = sigmoid(cs_prev * wcf + f)
+        ci = tanh(ci)
+
+        cs = ci .* i + cs_prev .* f
+        cs = clip(cs, cell_clip)
+
+        o = sigmoid(cs * wco + o)
+        co = tanh(cs)
+        h = co .* o
```

Comment:
can remove these math equations and give a link to https://github.com/tensorflow/tensorflow/blob/r1.8/tensorflow/contrib/rnn/python/ops/lstm_ops.py#L41-L114